lib/internal/modules/cjs/loader.js

```diff
@@ -608,8 +608,18 @@ Module._resolveFilename = function(request, parent, isMain, options) {
   // Look up the filename first, since that's the cache key.
   var filename = Module._findPath(request, paths, isMain);
   if (!filename) {
+    const requireChain = [];
+    for (var parentCursor = parent;
+      parentCursor;
+      parentCursor = parentCursor.parent) {
+      requireChain.push(parentCursor.filename || parentCursor.id);
+    }
+    let message = `Cannot find module '${request}'`;
+    if (requireChain.length > 0) {
+      message += ` required by:\n\t${requireChain.join('\n\t< ')}`;
```

Comment:
Can we also make this programmatically available on the `Error` object, so that we don’t tempt people to parse the error message?